src/api/controllers/ItemImageController.ts

```diff
@@ -18,46 +25,93 @@ export class ItemImageController {
 
     constructor(
         @inject(Types.Service) @named(Targets.Service.ItemImageService) private itemImageService: ItemImageService,
+        @inject(Types.Service) @named(Targets.Service.ListingItemTemplateService) private listingItemTemplateService: ListingItemTemplateService,
         @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType) {
         this.log = new Logger(__filename);
     }
 
-    @httpGet('/')
-    public async findAll( @response() res: myExpress.Response): Promise<any> {
-        const itemImages = await this.itemImageService.findAll();
-        this.log.debug('findAll: ', JSON.stringify(itemImages, null, 2));
-        return res.found(itemImages.toJSON());
-    }
-
-    @httpPost('/')
-    public async create( @response() res: myExpress.Response, @requestBody() body: any, @request() req: any): Promise<any> {
-
-        this.log.debug('files: ', req.files);
-//        const itemImage = await this.itemImageService.create(body);
-//        this.log.debug('create: ', JSON.stringify(itemImage, null, 2));
-//        return res.created(itemImage.toJSON());
-        return req.files[0];
-    }
-
-    @httpGet('/:id')
-    public async findOne( @response() res: myExpress.Response, @requestParam('id') id: string): Promise<any> {
-        const itemImage = await this.itemImageService.findOne(parseInt(id, 10));
-        this.log.debug('findOne: ', JSON.stringify(itemImage, null, 2));
-        return res.found(itemImage.toJSON());
-    }
-
-    @httpPut('/:id')
-    public async update( @response() res: myExpress.Response, @requestParam('id') id: string, @requestBody() body: any): Promise<any> {
-        const itemImage = await this.itemImageService.update(parseInt(id, 10), body);
-        this.log.debug('update: ', JSON.stringify(itemImage, null, 2));
-        return res.updated(itemImage.toJSON());
-    }
-
-    @httpDelete('/:id')
-    public async destroy( @response() res: myExpress.Response, @requestParam('id') id: string): Promise<any> {
-        await this.itemImageService.destroy(parseInt(id, 10));
-        this.log.debug('destroy: ', parseInt(id, 10));
-        return res.destroyed();
+    @httpPost('/template/:templateId')
+    public async create( @response() res: myExpress.Response, @requestParam('templateId') templateIdStr: string,
+                         @requestBody() body: any, @request() req: any): Promise<any> {
+        // check templateId is number
+        // also check listingItemTemplate id present in params
+        let templateId: number;
+        if (!templateIdStr) {
+            throw new MessageException('ListingItemTemplate id can not be null.');
+        } else {
+            templateId = parseInt(templateIdStr, 10);
+            if ( !templateId ) {
+                throw new MessageException('ListingItemTemplate id must be an integer.');
+            }
+        }
```

Comment:
pass the execution to service and do requestbody validation for values using annotations @IsNumber, @IsInt, @IsNotEmpty, etc and get rid of these unnecessary lines of code.